src/sidebar/components/MarkdownView.tsx

```diff
@@ -22,14 +48,49 @@ export default function MarkdownView({
   markdown,
   classes,
   style,
-  mentions,
+  mentions = [],
+  /* istanbul ignore next - test seam */
+  setTimeout_ = setTimeout,
+  /* istanbul ignore next - test seam */
+  clearTimeout_ = clearTimeout,
 }: MarkdownViewProps) {
   const html = useMemo(
     () => (markdown ? renderMathAndMarkdown(markdown) : ''),
     [markdown],
   );
   const content = useRef<HTMLDivElement | null>(null);
 
+  const mentionsPopoverAnchorRef = useRef<HTMLElement | null>(null);
+  const elementToMentionMap = useRef(
+    new Map<HTMLElement, Mention | InvalidUsername>(),
+  );
+  const [popoverContent, doSetPopoverContent] = useState<PopoverContent>(null);
+  /**
+   * This allows the content to be set with a small delay, so that popovers don't flickr
+   */
+  const popoverContentTimeout = useRef<ReturnType<typeof setTimeout> | null>();
+  const setPopoverContent = useCallback(
+    (content: PopoverContent) => {
+      if (popoverContentTimeout.current) {
+        clearTimeout_(popoverContentTimeout.current);
+      }
+
+      const setContent = () => {
+        doSetPopoverContent(content);
+        popoverContentTimeout.current = null;
+      };
+
+      // Set the content immediately when resetting, so that there's no delay
+      // when hiding the popover, only when showing it
+      if (content === null) {
+        setContent();
+      } else {
+        popoverContentTimeout.current = setTimeout_(setContent, 400);
```

Comment:
Popovers appear after hovering for 400ms. I compared the time it takes in GitHub, and this is the value I came up "by eye", because I wasn't able to find a way to properly measure it.